remote-messaging/remote-messaging-store/src/main/java/com/duckduckgo/remote/messaging/store/RemoteMessagingConfigRepository.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.remote.messaging.store
+
+import org.threeten.bp.LocalDateTime
+
+interface RemoteMessagingConfigRepository {
+    fun insert(remoteMessagingConfig: RemoteMessagingConfig)
+    fun get(): RemoteMessagingConfig
+    fun expired(): Boolean
+    fun invalidated(): Boolean
```

Comment:
These two methods are confusing to me, because they test the `RemoteMessagingConfig` model. They don't really belong to the repository imo.

Should we move this out, make them extension funs of `RemoteMessagingConfig` and call them `isExpired()` and `isInvalidated()` ?